src/scout_apm/core/core_agent_manager.py

```diff
@@ -53,77 +56,79 @@ def download(self):
 
     def run(self):
         try:
-            process = subprocess.Popen(
-                    self.agent_binary() +
-                    self.daemonize_flag() +
-                    self.log_level() +
-                    self.log_file() +
-                    self.config_file() +
-                    self.socket_path()
-                    )
-            process.wait()
+            subprocess.check_call(
+                self.agent_binary()
+                + self.daemonize_flag()
+                + self.log_level()
+                + self.log_file()
+                + self.config_file()
+                + self.socket_path()
+            )
```

Comment:
It probably does, but just verifying this is the same behavior as `Popen`+`wait`

Reply:
Oops — good catch, actually it isn't the same behavior if the process exits with a return code other than 0.

Before: `wait()` returns that code, which is ignored: `run()` returns `True`.

After: `check_call()` raises `CalledProcessError`, which is caught; `run()` returns `False`.

Which is the correct behavior? (Really it depends whether an exit code other than 0 signals a failure.)

Perhaps this change took care of `# TODO detect failure of launch properly`!

Reply:
A check (just to log failure) might be good. The goal is to have python try to start it (99% of cases it will), and if launch fails then we continue as normal. If there is no core agent running then we don't interrupt the python process at all but we will see log messages about not being able to communicate with it.

Reply:
Yes, the code does this. When launching the core agent fails and `check_call()` raises an exception, it's caught and logged:
```python
logger.error("Error running Core Agent: %r", e)
```